From 21edca582fcb6d703aaefa1ed7a56686b1185b69 Mon Sep 17 00:00:00 2001
From: Stephan Linz <linz@li-pro.net>
Date: Fri, 18 Nov 2011 17:39:08 +0100
Subject: [PATCH] FDT: fix dependency to LMB support

Architectures without LMB support (undef CONFIG_LMB) will break
compilation on bootm comand and image handling. So we need more
conditions to avoid undefined or unresolved symbols.

This affects arch/microblaze.

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 common/cmd_bootm.c |    2 +-
 common/image.c     |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/cmd_bootm.c b/common/cmd_bootm.c
index 8909ee7..6c44e4b 100644
--- a/common/cmd_bootm.c
+++ b/common/cmd_bootm.c
@@ -547,7 +547,7 @@ int do_bootm_subcommand (cmd_tbl_t *cmdtp, int flag, int argc, char * const argv
 		}
 			break;
 #endif
-#if defined(CONFIG_OF_LIBFDT)
+#if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_LMB)
 		case BOOTM_STATE_FDT:
 		{
 			boot_fdt_add_mem_rsv_regions(&images.lmb,
diff --git a/common/image.c b/common/image.c
index d38ce4a..2d15cd1 100644
--- a/common/image.c
+++ b/common/image.c
@@ -1179,7 +1179,7 @@ static int fit_check_fdt (const void *fit, int fdt_noffset, int verify)
 #define CONFIG_SYS_FDT_PAD 0x3000
 #endif
 
-#if defined(CONFIG_OF_LIBFDT)
+#if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_LMB)
 /**
  * boot_fdt_add_mem_rsv_regions - Mark the memreserve sections as unusable
  * @lmb: pointer to lmb handle, will be used for memory mgmt
-- 
1.7.0.4

